### DISCUSSION TITLE
[Bug] x-bind:class handlers are lost on Turbo Drive restoration

### DISCUSSION THREAD

**Discussion by davidalejandroaguilar** [alpinejs/alpine#3485] [7. Bugs (archived)]
Whenever a Turbo Drive restoration happens, `x-bind:class` handlers are lost, needing to reload the page. This does not happen on `x-show`.

[Here's a new Rails app](https://github.com/davidalejandroaguilar/alpinejs-bind-class-bug) reproducing the issue. [Here's the relevant commit](https://github.com/davidalejandroaguilar/alpinejs-bind-class-bug/commit/7446f076f66a1d8462f4e830c393387d97e7ca55).

Also attaching a video:

https://user-images.githubusercontent.com/15317732/227666441-0e9c16db-2dd2-40b0-97a0-22bbeb5535b0.mov

**Comment by ekwoka** [comment]
🤔 curious indeed

Do you have a link to this Turbo Drive thing?

**Reply by davidalejandroaguilar** [reply]
Sure thing, here's the documentation https://turbo.hotwired.dev/handbook/drive. Here's the JS package https://github.com/hotwired/turbo. 

I think whatever event handling logic `x-show` has should also be applied to `x-bind`? (e.g. an observer).

**Reply by ekwoka** [reply]
Okay, yeah I'm aware of that.

It should just work, they both basically function the same way.

I have seen issues with elements not properly handling being swapped out live on the page.... like the mutation observer isn't properly seeing all of the added nodes...
